### PR TITLE
Add docs for mysql `caching_sha2_password` default

### PR DESCRIPTION
**Pull request by huwd**
This stumped me for a couple of days.

Since the mysql 8.0 update I had been unable to `make whitehall` seeing errors when i tried to run: `govuk-docker run whitehall-lite bundle exec rails db:prepare`.

The stack trace was as follows:

```
ActiveRecord::ConnectionNotEstablished: Plugin caching_sha2_password could not be loaded: /usr/lib/x86_64-linux-gnu/mariadb19/plugin/caching_sha2_password.so: cannot open shared object file: No such file or dire
ctory                                                                                                                                                                                                              
/govuk/whitehall/app/models/edition/translatable.rb:19:in `block in <module:Translatable>'                                                                                                                         
/govuk/whitehall/app/models/edition.rb:23:in `include'                                                                                                                                                             
/govuk/whitehall/app/models/edition.rb:23:in `<class:Edition>'                                                                                                                                                     
/govuk/whitehall/app/models/edition.rb:3:in `<main>'                                                                                                                                                               
/govuk/whitehall/app/models/publicationesque.rb:9:in `<main>'                                                                                                                                                      
/govuk/whitehall/app/models/publication.rb:8:in `<main>'                                                                                                                                                           
/govuk/whitehall/app/helpers/admin/edition_routes_helper.rb:4:in `<module:EditionRoutesHelper>'                                                                                                                    
/govuk/whitehall/app/helpers/admin/edition_routes_helper.rb:1:in `<main>'                                                                                                                                          
/govuk/whitehall/config/initializers/teaspoon.rb:2:in `<main>'                                                                                                                                                     
/govuk/whitehall/config/environment.rb:5:in `<main>'                                                                                                                                                               
                                                                                                                                                                                                                   
Caused by:                                                                                                                                                                                                         
Mysql2::Error::ConnectionError: Plugin caching_sha2_password could not be loaded: /usr/lib/x86_64-linux-gnu/mariadb19/plugin/caching_sha2_password.so: cannot open shared object file: No such file or directory   
/govuk/whitehall/app/models/edition/translatable.rb:19:in `block in <module:Translatable>'                                                                                                                         
/govuk/whitehall/app/models/edition.rb:23:in `include'                                                                                                                                                             
/govuk/whitehall/app/models/edition.rb:23:in `<class:Edition>'                                                                                                                                                     
/govuk/whitehall/app/models/edition.rb:3:in `<main>'                                                                                                                                                               
/govuk/whitehall/app/models/publicationesque.rb:9:in `<main>'                                                                                                                                                      
/govuk/whitehall/app/models/publication.rb:8:in `<main>'                                                                                                                                                           
/govuk/whitehall/app/helpers/admin/edition_routes_helper.rb:4:in `<module:EditionRoutesHelper>'                                                                                                                    
/govuk/whitehall/app/helpers/admin/edition_routes_helper.rb:1:in `<main>'                                                                                                                                          
/govuk/whitehall/config/initializers/teaspoon.rb:2:in `<main>'                                                                                                                                                     
/govuk/whitehall/config/environment.rb:5:in `<main>'                                                                                                                                                               
Tasks: TOP => db:prepare => db:load_config => environment                                                                                                                                                          
(See full trace by running task with --trace)
```

This seems to be related to a change in [MySQL password default config][1] in v8.
This documents a solution, (with massive thanks to @ChrisBAshton and @brucebolt)

We may want to find a more sustainable solution as part of the MySQL docker setup. 
[I have created an issue here for that](https://github.com/alphagov/govuk-docker/issues/551).

[1]: https://dev.mysql.com/doc/refman/8.0/en/caching-sha2-pluggable-authentication.html

### CODE REVIEW

File: docs/troubleshooting.md
Comment (alex9smith) [inline docs/troubleshooting.md:138]: Could you add an example of an app that uses MySQL?

**Review by alex9smith — APPROVED** [review]
LGTM, one non-blocking request

File: docs/troubleshooting.md
Comment (ChrisBAshton) [inline docs/troubleshooting.md:117]:
```suggestion
## Cannot `rails db:prepare` or start console due to "Plugin caching_sha2_password could not be loaded"
```

File: docs/troubleshooting.md
Comment (ChrisBAshton) [inline docs/troubleshooting.md:124]:
```suggestion
```

File: docs/troubleshooting.md
Comment (ChrisBAshton) [inline docs/troubleshooting.md:136]: We can make it less wordy:

```suggestion
A workaround is to get MySQL to fall back to using `mysql_native_password` as follows:
```

**Review by ChrisBAshton — COMMENTED** [review]
Thanks a lot for getting this documented!

I'm not sure it needs quite such a detailed introduction (perhaps some of the context can be added to the commit message itself, rather than the doc), as most devs will just want to dive in and copy and paste the fixes. But happy to leave it in if it's useful 👍

**Comment by huwd** [comment]
Great thanks both, will get those amended today

File: docs/troubleshooting.md
Comment (huwd) [reply docs/troubleshooting.md:124]: woops

File: docs/troubleshooting.md
Comment (ChrisBAshton) [inline docs/troubleshooting.md:131]: Nitpick (as we're already in the console at this point):

```suggestion
- Alter the way the root user identifies itself. `ALTER USER 'root' IDENTIFIED WITH mysql_native_password BY 'root';`
```

**Review by ChrisBAshton — APPROVED** [review]
LGTM 👍  one non-blocking comment if you fancy addressing before merging. Shouldn't need  a re-review from me.

File: docs/troubleshooting.md
Comment (huwd) [reply docs/troubleshooting.md:131]: Ah yeah, thought about that at the time, was shooting for "don't punch this into bash"- But you are right on reflection, a dev just wants a copy and paste and this'll get in the way.

**Pull request merged**
